### PR TITLE
Improved Power On Reset  Issue #1808

### PR DESCRIPTION
**Pull request by mbaillif**
improved Power On Reset and additional French translation

### CODE REVIEW

File: src/main/java/com/cburch/logisim/std/wiring/PowerOnReset.java [hunk 1]
```diff
@@ -103,50 +186,123 @@ public void actionPerformed(ActionEvent e) {
...
+      
+      x1 = x + 4;
+      x2 = x + width / 2;
+      x3 = x + width - 4;
+      y1 = y + offset + 2;
+      y2 = y + height - 5;
+
+      final var pstat = painter.getAttributeValue(PORTRANS); 
+      if (pstat == LTOH) {
+        y1 = y1 ^ y2 ^ (y2 = y1);  // swap y1 <-> y2
```
Comment (davidhutchens) [inline src/main/java/com/cburch/logisim/std/wiring/PowerOnReset.java:293]: Write this as:
var tmp = y1;
y1 = y2;
y2 = tmp;

It is better not to use unusual methods. Your expression leaves questions about the value of y2 when it is fetched. The optimizer will find the tmp easier to optimize.

File: src/main/resources/resources/logisim/strings/std/std.properties [hunk 2]
```diff
@@ -896,8 +896,16 @@ pinThreeStateAttr = Three-state?
...
 # wiring/PowerOnReset.java
 #
-porHighAttr = POR-time
 PowerOnResetComponent = POR
+porLongName = Power-On Reset
+porHighAttr = POR-time
+PorSize = Draw size
+porMediumSize = Medium
+porNarrowSize = Narrow
+porOldSize = Old size
```
Comment (davidhutchens) [inline src/main/resources/resources/logisim/strings/std/std.properties:905]: Wide

File: src/main/resources/resources/logisim/strings/std/std.properties [hunk 3]
```diff
@@ -896,8 +896,16 @@ pinThreeStateAttr = Three-state?
...
+porTransition = Port transition
+porHightToLow = Hight to Low
```
Comment (davidhutchens) [inline src/main/resources/resources/logisim/strings/std/std.properties:907]: High to Low

File: src/main/resources/resources/logisim/strings/std/std.properties [hunk 4]
```diff
@@ -896,8 +896,16 @@ pinThreeStateAttr = Three-state?
...
+porLowToHight = Low to Hight
```
Comment (davidhutchens) [inline src/main/resources/resources/logisim/strings/std/std.properties:908]: Low to High

**Review by davidhutchens — COMMENTED** [review]
I have made some minor suggestions.

File: src/main/java/com/cburch/logisim/std/wiring/PowerOnReset.java (hunk 1 above)
Comment (mbaillif) [reply src/main/java/com/cburch/logisim/std/wiring/PowerOnReset.java:293]: Ok, it's true that even for humans it's more understandable :-))

**Review by davidhutchens — APPROVED** [review]
Looks good to me.

**Pull request merged**